File: kingfisher-web/src/api/worktask.ts
import request from './request';

export interface WorkTask {
  readonly id: number;
  readonly title: string;
  readonly description: string;
  readonly owner_id: number;
  readonly department_id: number;
  readonly status: string;
  readonly created_at: string;
  readonly updated_at: string;
}

export interface WorkTaskInput {
  readonly title: string;
  readonly description: string;
  readonly department_id: number;
  readonly status: string;
}

export const workTaskApi = {
  list: (params: Record<string, unknown>) => request.get('/tasks', { params }),
  getById: (id: number) => request.get(`/tasks/${id}`),
  create: (data: WorkTaskInput) => request.post('/tasks', data),
  update: (id: number, data: WorkTaskInput) => request.put(`/tasks/${id}`, data),
  delete: (id: number) => request.delete(`/tasks/${id}`),
};

File: java/src/main/java/com/kingfisher/modules/worktask/domain/WorkTask.java
package com.kingfisher.modules.worktask.domain;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 工作任务领域实体，对应 tasks 表。
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class WorkTask {

    private Long id;
    private String title;
    private String description;
    @JsonProperty("owner_id")
    private Long ownerId;
    @JsonProperty("department_id")
    private Long departmentId;
    private String status;
    @JsonProperty("created_at")
    private LocalDateTime createdAt;
    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;
}
